src/foam/u2/TextFormatter.js

```diff
@@ -0,0 +1,282 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.INTERFACE({
+  package: 'foam.u2',
+  name: 'TextFormatter',
+  properties: [
+    {
+      name: 'formatter',
+      class: 'Object'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'getPlaceholder',
+      args: [
+        { name: 'formattedData', type: 'String' }
+      ],
+      documentation: `
+        Calculate a placeholder text for the FormattedTextField
+        If formattedData is provided, calculate a dynamic placeholder based on existing data
+        Otherwise, return the default placeholder text
+      `
+    },
+    {
+      name: 'getMaxLength',
+      documentation: `
+        Returns the max-length of the FormattedTextField based on allowable input length plus formatting
+      `
+    },
+    {
+      name: 'getUnformatted',
+      args: [
+        { name: 'formattedData', type: 'String' }
+      ],
+      documentation: `
+        Returns the unformatted version of a formatted String
+      `
+    },
+    {
+      name: 'onDelete',
+      args: [
+        { name: 'data', type: 'String' },
+        { name: 'start', type: 'Int' },
+        { name: 'end', type: 'Int' }
+      ],
+      documentation: `
+        Configure properties for delete, computes and returns a new selection range to
+        delete from
+      `
+    },
+    {
+      name: 'formatData',
+      args: [
```

Comment:
Okay to keep it this way, but you could just do:
args: 'String data, String old, Int startingPos, Int endPos, Boolean isDelete'